pkg/helpers/helpers.go

```diff
@@ -728,6 +744,18 @@ func FindRelatedResourcesStatus(
 	return nil
 }
 
+func RemoveRelatedResourcesStatus(
+	relatedResourcesStatuses *[]operatorapiv1.RelatedResourceMeta,
+	relatedResource operatorapiv1.RelatedResourceMeta) {
+	var result []operatorapiv1.RelatedResourceMeta
+	for _, v := range *relatedResourcesStatuses {
+		if !reflect.DeepEqual(v, relatedResource) {
```

Comment:
relatedResource is a plain struct that you can directly use == I think

Reply:
fixed